OpenAI_API/OpenAIAPI.cs

```diff
@@ -1,42 +1,68 @@
-		public OpenAIAPI(APIAuthentication apiKeys = null)
-		{
-			this.Auth = apiKeys.ThisOrDefault();
-			Completions = new CompletionEndpoint(this);
-			Models = new ModelsEndpoint(this);
-			Files = new FilesEndpoint(this);
-			Embeddings = new EmbeddingEndpoint(this);
-		}
+        /// <summary>The HTTP client configured with the authentication headers.</summary>
+        internal HttpClient Client { get; }
 
-		/// <summary>
-		/// Text generation is the core function of the API. You give the API a prompt, and it generates a completion. The way you “program” the API to do a task is by simply describing the task in plain english or providing a few written examples. This simple approach works for a wide range of use cases, including summarization, translation, grammar correction, question answering, chatbots, composing emails, and much more (see the prompt library for inspiration).
-		/// </summary>
-		public CompletionEndpoint Completions { get; }
+        private OpenAIAPI()
+        {
+            this.Completions = new CompletionEndpoint(this);
+            this.Models = new ModelsEndpoint(this);
+            this.Files = new FilesEndpoint(this);
+            this.Embeddings = new EmbeddingEndpoint(this);
+        }
+
+        /// <summary>
+        /// Creates a new entry point to the OpenAPI API, handling auth and allowing access to the various API endpoints
+        /// </summary>
+        /// <param name="httpClient">The HTTP client configured with the authentication headers.</param>
+        public OpenAIAPI(HttpClient httpClient) : this() =>
+            this.Client = httpClient;
+
+        /// <summary>
+        /// Creates a new entry point to the OpenAPI API, handling auth and allowing access to the various API endpoints
+        /// </summary>
+        /// <param name="auth">The authentication details for the API</param>
+        [Obsolete("""
+        This constructor will generate a new HTTP client every time it is called.
+        Do not use this in scenarios where multiple instances of the API are required.
+        This is provided for backwards compatibility, use .NET Dependency Injection instead.
+        """, false)]
+        public OpenAIAPI(APIAuthentication auth) : 
+            this(EndpointBase.ConfigureClient(new HttpClient(), auth)) { }
+
+        /// <summary>
+        /// Creates a new entry point to the OpenAPI API, handling auth and allowing access to the various API endpoints
+        /// </summary>
+        /// <param name="key">The authentication key for the API</param>
+        [Obsolete("""
+        This constructor will generate a new HTTP client every time it is called.
+        Do not use this in scenarios where multiple instances of the API are required.
+        This is provided for backwards compatibility, use .NET Dependency Injection instead.
+        """, false)]
+        public OpenAIAPI(string key) :
+            this(EndpointBase.ConfigureClient(new HttpClient(), new APIAuthentication(key))) { }
```

Comment:
An idea to give you the backwards compatibility, but it's somewhat messy - alternate constructors that create an `HttpClient` when called and configure it, but that match the old patterns.

These will cause warnings for users upgrading still.